permissions/__main__.py

```diff
@@ -25,10 +25,12 @@ async def permission(request, username):
     else:
         post_data = await request.post() # request.json() crashes on empty data
         LOG.debug('POST DATA: %s', post_data)
-        try:
-            requested_datasets = post_data.get('datasets', '').split(',')
-        except:
-            requested_datasets = list(post_data.get('datasets', ''))
+        if post_data.get('datasets') is None:
```

Comment:
Move `post_data.get('datasets')` upwards into its own variable.
Otherwise each call to .get will run a search in the hashtable.